### PR TITLE
chore(api!): move additional methods to light client module

### PR DESCRIPTION
**Pull request by TropicalDog17**
## Description

closes: #6087 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **Refactor**
	- Improved the logic handling and organization of client states by moving specific functionalities to the Light Client Module.
- **Documentation**
	- Updated migration guide to reflect the changes in client state management.
- **Bug Fixes**
	- Enhanced stability by ensuring the client status accurately reflects the frozen state when applicable.

### CODE REVIEW

**Review by crodriguezvega — APPROVED** [review]
Thank you for picking up this one, @TropicalDog17!

File: modules/light-clients/06-solomachine/light_client_module.go [hunk 1]
```diff
@@ -164,7 +167,10 @@ func (l LightClientModule) VerifyNonMembership(
 	return clientState.VerifyNonMembership(ctx, clientStore, l.cdc, height, delayTimePeriod, delayBlockPeriod, proof, path)
 }
 
-// Status obtains the client state associated with the client identifier and calls into the clientState.Status method.
+// Status returns the status of the solo machine client.
+// The client may be:
+// - Active: if frozen sequence is 0
+// - Frozen: otherwise solo machine is frozen
```
Comment (damiannolan) [inline modules/light-clients/06-solomachine/light_client_module.go:173]: I think this was a mistake in the original godoc for solomachine. Frozen sequence is used in tendermint clients, but solomachine clients only have the `IsFrozen` bool on the client state.

Comment (damiannolan) [reply modules/light-clients/06-solomachine/light_client_module.go:173]:
```suggestion
// The client may be:
// - Active: if `IsFrozen` is false.
// - Frozen: if `IsFrozen` is true.
// - Unknown: if the client state associated with the provided client identifier is not found.
```

**Review by damiannolan — APPROVED** [review]
Thank you!

File: CHANGELOG.md [hunk 2]
```diff
@@ -47,6 +47,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (core/02-client, light-clients) [\#5806](https://github.com/cosmos/ibc-go/pull/5806) Decouple light client routing from their encoding structure.
 * (core/04-channel) [\#5991](https://github.com/cosmos/ibc-go/pull/5991) The client CLI `QueryLatestConsensusState` has been removed.
 * (light-clients/06-solomachine) [\#6037](https://github.com/cosmos/ibc-go/pull/6037) Remove `Initialize` function from `ClientState` and move logic to `Initialize` function of `LightClientModule`.
+* (light-clients/06-solomachine) [\#6230](https://github.com/cosmos/ibc-go/pull/6230) Remove `GetTimestampAtHeight`, `Status` and `UpdateStateOnMisbehaviour` functions from `ClientState` and move logic to functions of `LightClientModule`.
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply CHANGELOG.md:50]: all of the changelogs follow the existing format

**Pull request merged**